lib/Crypto/decryptrequest.cpp

```diff
@@ -252,13 +310,22 @@ void DecryptRequest::startRequest()
             emit resultChanged();
         }
 
-        QDBusPendingReply<Result, QByteArray> reply =
-                d->m_manager->d_ptr->decrypt(d->m_data,
-                                             d->m_initialisationVector,
-                                             d->m_key,
-                                             d->m_blockMode,
-                                             d->m_padding,
-                                             d->m_cryptoPluginName);
+        QDBusPendingReply<Result, QByteArray> reply = (d->m_blockMode == CryptoManager::BlockModeGcm)
```

Comment:
I wonder about this.  One of the requirements from OMP is that we must support arbitrary plugin-specific block modes (i.e. custom algorithm, custom block mode; see CryptoManager::BlockModeCustom).  It could be that such a block mode is authenticated even though it isn't one of GCM or CCM, and thus we must not switch the service call based on this.

That is: instead of adding a separate "authenticatedDecrypt" (and authenticatedEncrypt) which we call if and only if the block mode is a "known" authenticated mode, we should probably simply expand the "normal" encrypt/decrypt APIs so that they take/return the authenticated operation data, and simply document that those extra authenticated operation data are not valid for non-authenticated block modes (such as GCM/CCM).

Reply:
What if this if condition is changed to check for whether authenticationData is set, rather than checking the block mode? Is that sufficient to support the arbitrary block mode requirement?

Whether we should have both encrypt() and authenticatedEncrypt() methods in the daemon/plugins can be considered separately; I think having a separate authenticatedEncrypt() is useful to be explicit about the fact that the authentication data and tag are required for the auth case.